Jenkinsfile

```diff
@@ -30,17 +30,48 @@ pipeline {
         //         }
         //     }
         // }
-        stage('Go build') {
+        stage('Go build x64') {
             agent {
                 docker {
-                    image docker.build("storj-ci", "--pull https://github.com/storj/ci.git#main").id
-                    args '--user root:root --volume "/tmp/gomod":/go/pkg/mod '
+                    image 'golang:1.17.2'
+                    args "--volume /tmp/gomod:/go/pkg/mod --user root:root"
                 }
             }
             steps {
                 script {
-                    sh './build.sh'
-                    stash(name: "build", includes: "build/")
+                    // get owner UID of working directory to run commands as that user
+                    def userId = sh(script: "stat -c '%u' .", returnStdout: true).trim()
+                    sh "useradd --create-home --uid ${userId} jenkins"
+
+                    sh 'su jenkins -c "make build-x64"'
+                    stash(name: "build-x64", includes: "build/")
+                }
+            }
+            post {
+                always {
+                    cleanWs()
```

Comment:
`post` code isn't guaranteed to run due to issues within Jenkins

Reply:
Stefan showed me that cleanup is done by https://plugins.jenkins.io/resource-disposer/

What do you suggest? I don't see harm in leaving `cleanWs()` here.

Reply:
storj/storj does a delete before running: https://github.com/storj/storj/blob/main/Jenkinsfile.public#L23

Reply:
storj/storj is forced to do that, because it leaves behind files in the workspace with root owner, which can only be removed by the root user. 

After this PR uplink-php doesn't have that problem. Normal cleanup should work.

Reply:
The cleanup failed due to a timeout / jenkins crash not due to the permissions part.